### PR TITLE
Set Block If queue full to false by default

### PR DESCRIPTION
**Pull request by jai1**
### Motivation

Fix for #422

### Modifications

- Changed default options in CPP and Java Client ProducerConfigurations
- Fixed some test cases
- Changed perf tests to block if queue is full

### Result

sendAsync becomes a non-blocking call

### Do not merge this

### CODE REVIEW

**Comment by msb-at-yahoo** [comment]
how is the default queue size change related?

**Comment by jai1** [comment]
In production, if we have a producer with the following config:-
a. Queue size 1K
b. Produce rate = 2K msg/s

And lets say the broker has a GC pause of 500 milliseconds or a rolling restart is going on then:- 
a. With current code base the sendAsync will block for few milliseconds and client will not even know that there was some issue (except high latency)
b. After this PR the client will start seeing failures and will come to the conclusion that the service is down or not working. 

So in order to prevent clients seeing failures, we assumed that most properties will not produce over 1K MPS and so a queue size of 30K will prevent them from seeing failures before 30 seconds (default timeout). 

But yes I expect there to be some discussion regarding the queue size before we the PR is merged 

@merlimat @saandrews @rdhabalia  - do you agree with this approach?

**Comment by msb-at-yahoo** [comment]
i think it'd be better to separate the queue size change from the block if full change. both might make sense, but they're not related.

File: pulsar-client/src/main/java/com/yahoo/pulsar/client/api/ProducerConfiguration.java [hunk 1]
```diff
@@ -36,8 +36,8 @@
      */
     private static final long serialVersionUID = 1L;
     private long sendTimeoutMs = 30000;
-    private int maxPendingMessages = 1000;
-    private boolean blockIfQueueFull = true;
+    private int maxPendingMessages = 30000;
+    private boolean blockIfQueueFull = false;
```
Comment (merlimat) [inline pulsar-client/src/main/java/com/yahoo/pulsar/client/api/ProducerConfiguration.java]: We should also update the Javadoc on the getter/setter

**Comment by jai1** [comment]
@merlimat - Updated the docs
@msb-at-yahoo - Moving the maxPendingMessages changes to separate PR  #445

**Comment by jai1** [comment]
@saandrews - Have made relevant change to Open Source documentation, will update Yahoo wrapper documentation and release notes once the PR is merged.

**Pull request merged**